### PR TITLE
Add extended cell kinds: state, concurrent, universe-builder, trade

### PR DESCRIPTION
**Pull request by rodoHasArrived**
## Description

This PR extends the strategy builder with four new cell kinds that enable more sophisticated strategy modeling:

- **State cells** (`state`): Named execution states with entry/exit conditions for lifecycle state machine modeling
- **Concurrent cells** (`concurrent`): Parallel evaluation points with configurable pass semantics (all-pass, any-pass, first-wins)
- **Universe-builder cells** (`universe-builder`): Structured asset class and rule-based universe construction with size constraints
- **Trade cells** (`trade`): Explicit trade intent definition with instrument type, direction, sizing method, and price constraints

Each new cell kind includes:
- Type-safe parameter interfaces (`StateCellParameters`, `ConcurrentCellParameters`, `UniverseBuilderCellParameters`, `TradeCellParameters`)
- Typed parameter extraction functions (`getStateCellParams`, `getConcurrentCellParams`, `getUniverseBuilderCellParams`, `getTradeCellParams`)
- Validation logic in both TypeScript and C# to enforce required parameters and valid enum values
- Four new template strategies demonstrating each cell kind in realistic workflows
- Inspector UI that surfaces typed parameters as structured fields with appropriate tone/styling
- Canvas rendering with kind-specific icons and badges

## Type of Change

- New feature
- Code refactoring
- Test improvements

## Motivation and Context

The strategy builder previously supported only basic cell kinds (visual, formula, code, governance, options-payoff). These new cell kinds enable:

1. **State machines**: Model strategies as lifecycle state machines with named checkpoints and conditional transitions
2. **Parallel signal evaluation**: Run multiple signal branches concurrently with explicit pass semantics
3. **Disciplined universe construction**: Define asset class, inclusion/exclusion rules, and size bounds explicitly
4. **Explicit trade execution**: Specify instrument, direction, sizing method, and price constraints as first-class strategy elements

The implementation maintains backward compatibility while adding structured parameter support to the `StrategyBuilderCell` interface.

## How Has This Been Tested?

- Added 10 new unit tests covering parameter validation, extraction, template loading, and UI rendering
- Added 3 new integration tests verifying template loading and inspector field population
- Added 5 new C# validation tests covering parameter requirements and edge cases
- All existing tests continue to pass
- Four new template strategies included with full validation coverage

## Checklist

- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] I have checked my code and corrected any misspellings
- [x] My changes generate no new warnings

## Related Issues

None

## Additional Notes

The implementation is split across:
- `src/Meridian.Ui/dashboard/src/screens/strategy-designer-screen.view-model.ts` - TypeScript types, validators, extractors, templates, and UI view model builders
- `src/Meridian.Strategies/Services/StrategyDesignService.cs` - C# validation and template definitions
- Test files with comprehensive coverage of validation, extraction, and UI rendering

All parameter validation is performed consistently in both TypeScript and C# to ensure parity between client and server validation.

https://claude.ai/code/session_01ULHzoenDsS44aY275BbZph

### CODE REVIEW

File: src/Meridian.Ui/dashboard/src/screens/strategy-designer-screen.view-model.ts [hunk 1]
```diff
@@ -924,6 +1210,60 @@ export function validateStrategyBuilderDocument(
...
+    if (!dir || !validDirections.includes(dir)) {
+      messages.push({ code: "TradeCellDirectionRequired", severity: "error", targetId: cell.cellId, message: `${label} (trade) direction must be Buy, Sell, SellShort, or BuyToCover.` });
+    }
+    if (!sizing || !validSizing.includes(sizing)) {
+      messages.push({ code: "TradeCellSizingRequired", severity: "error", targetId: cell.cellId, message: `${label} (trade) sizingMethod must be FixedShares, FixedNotional, PercentAUM, or EqualWeight.` });
+    }
+  }
+
+  return messages;
+}
```
[the remark this answers is not in the document]
Comment (rodoHasArrived) [reply src/Meridian.Ui/dashboard/src/screens/strategy-designer-screen.view-model.ts]: @copilot correct this issue

**Comment by rodoHasArrived** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/rodoHasArrived/Meridian-main/pull/1298#pullrequestreview-4300887864)

File: src/Meridian.Ui/dashboard/src/screens/strategy-designer-screen.view-model.ts [hunk 2]
```diff
@@ -924,6 +1210,60 @@ export function validateStrategyBuilderDocument(
...
+function validateCellKindParameters(cell: StrategyBuilderCell): StrategyBuilderValidationMessage[] {
+  const messages: StrategyBuilderValidationMessage[] = [];
+  const label = cell.label || cell.cellId;
+
+  if (cell.kind === "state") {
+    if (!cell.parameters?.["exitCondition"]) {
+      messages.push({ code: "StateCellExitRequired", severity: "error", targetId: cell.cellId, message: `${label} (state) must define an exitCondition parameter.` });
+    }
+    if (!cell.parameters?.["stateLabel"]) {
+      messages.push({ code: "StateCellLabelRequired", severity: "error", targetId: cell.cellId, message: `${label} (state) must define a stateLabel parameter.` });
```
[the remark this answers is not in the document]
Comment (rodoHasArrived) [reply src/Meridian.Ui/dashboard/src/screens/strategy-designer-screen.view-model.ts]: @copilot fix this

File: src/Meridian.Ui/dashboard/src/screens/strategy-designer-screen.view-model.ts [hunk 3]
```diff
@@ -924,6 +1210,60 @@ export function validateStrategyBuilderDocument(
...
+  }
+
+  if (cell.kind === "concurrent") {
+    if (!cell.parameters?.["branchIds"]) {
+      messages.push({ code: "ConcurrentCellBranchesRequired", severity: "error", targetId: cell.cellId, message: `${label} (concurrent) must define a branchIds parameter listing the parallel cell IDs.` });
+    }
+    const validSemantics: ConcurrentCellSemantics[] = ["all-pass", "any-pass", "first-wins"];
+    const sem = cell.parameters?.["semantics"];
+    if (!sem || !validSemantics.includes(sem as ConcurrentCellSemantics)) {
+      messages.push({ code: "ConcurrentCellSemanticsRequired", severity: "error", targetId: cell.cellId, message: `${label} (concurrent) semantics must be one of: all-pass, any-pass, first-wins.` });
```
[the remark this answers is not in the document]
Comment (rodoHasArrived) [reply src/Meridian.Ui/dashboard/src/screens/strategy-designer-screen.view-model.ts]: @copilot fix this

File: src/Meridian.Strategies/Services/StrategyDesignService.cs [hunk 4]
```diff
@@ -430,12 +432,69 @@ private static StrategyDesignValidationMessage Warning(string code, string targe
...
+
+                if (p is null || !p.TryGetValue("stateLabel", out var stateLbl) || string.IsNullOrWhiteSpace(stateLbl))
+                    yield return Error("StateCellLabelRequired", cell.CellId, $"{label} (state) must define a stateLabel parameter.");
+                break;
+
+            case "concurrent":
+                if (p is null || !p.TryGetValue("branchIds", out var branches) || string.IsNullOrWhiteSpace(branches))
+                    yield return Error("ConcurrentCellBranchesRequired", cell.CellId, $"{label} (concurrent) must define a branchIds parameter.");
+
+                var validSemantics = new[] { "all-pass", "any-pass", "first-wins" };
```
[the remark this answers is not in the document]
Comment (rodoHasArrived) [reply src/Meridian.Strategies/Services/StrategyDesignService.cs]: @copilot fix this

File: src/Meridian.Strategies/Services/StrategyDesignService.cs [hunk 5]
```diff
@@ -430,12 +432,69 @@ private static StrategyDesignValidationMessage Warning(string code, string targe
...
-            ? $"{cell.Kind} cell runs without catalog fields."
-            : $"{cell.Kind} cell uses {string.Join(", ", refs)}.";
+        var refsLabel = refs.Count == 0 ? "no catalog fields" : string.Join(", ", refs);
+        var p = cell.Parameters;
+        return cell.Kind switch
+        {
+            "state" => $"State '{p?.GetValueOrDefault("stateLabel", cell.CellId)}' exits on: {p?.GetValueOrDefault("exitCondition", "(none)")}.",
+            "concurrent" => $"Concurrent evaluation ({p?.GetValueOrDefault("semantics", "?")}) across: {p?.GetValueOrDefault("branchIds", "(none)")}.",
+            "universe-builder" => $"Builds {p?.GetValueOrDefault("assetClass", "?")} universe using {refsLabel}.",
+            "trade" => $"Trade intent: {p?.GetValueOrDefault("direction", "?")} {p?.GetValueOrDefault("instrument", "?")} via {p?.GetValueOrDefault("sizingMethod", "?")}.",
```
[the remark this answers is not in the document]
Comment (rodoHasArrived) [reply src/Meridian.Strategies/Services/StrategyDesignService.cs:625]: @copilot fix this

File: src/Meridian.Ui/dashboard/src/screens/strategy-designer-screen.view-model.ts [hunk 6]
```diff
@@ -187,6 +200,99 @@ export interface StrategyBuilderCell {
...
+    stateLabel,
+    isTerminal: cell.parameters["isTerminal"]
+  };
+}
+
+export function getConcurrentCellParams(cell: StrategyBuilderCell): ConcurrentCellParameters | null {
+  if (cell.kind !== "concurrent" || !cell.parameters) return null;
+  const branchIds = cell.parameters["branchIds"];
+  const semantics = cell.parameters["semantics"] as ConcurrentCellSemantics | undefined;
+  const validSemantics: ConcurrentCellSemantics[] = ["all-pass", "any-pass", "first-wins"];
```
[the remark this answers is not in the document]
Comment (rodoHasArrived) [reply src/Meridian.Ui/dashboard/src/screens/strategy-designer-screen.view-model.ts]: @copilot fix this
